lib/tldr.js

```diff
@@ -59,138 +58,107 @@ class Tldr {
       .then((content) => {
         // Getting the shortindex first to populate the shortindex var
         return index.getShortIndex().then(() => {
           this.renderContent(content);
         });
-      })
-      .catch((err) => {
-        console.error(err);
-        exit(1);
       });
   }
 
   clearCache() {
-    this.cache.clear().then(() => {
+    return this.cache.clear().then(() => {
       console.log('Done');
     });
   }
 
   updateCache() {
-    const spinner = ora();
-    spinner.start('Updating...');
-    return this.cache.update()
-      .then(() => {
-        spinner.succeed();
-      })
-      .catch((err) => {
-        console.error(err);
-        exit(1);
-      });
+    return spinningPromise('Updating...', () => {
+      return this.cache.update();
+    });
   }
 
   updateIndex() {
-    const spinner = ora();
-    spinner.start('Creating index...');
-    search.createIndex()
-      .then(() => {
-        spinner.succeed();
-      })
-      .catch((err) => {
-        console.error(err);
-      });
+    return spinningPromise('Creating index...', () => {
+      return search.createIndex();
+    });
   }
 
   search(keywords) {
-    search.getResults(keywords.join(' '))
+    return search.getResults(keywords.join(' '))
       .then((results) => {
         // TODO: make search into a class also.
         search.printResults(results, this.config);
-      })
-      .catch((err) => {
-        console.error(err);
       });
   }
 
   printPages(pages, singleColumn) {
     if (pages.length === 0) {
-      console.error(messages.emptyCache());
-      exit(1);
+      throw new Error(messages.emptyCache());
     }
-    this.checkStale();
-    let endOfLine = require('os').EOL;
-    let delimiter = singleColumn ? endOfLine : ', ';
-    console.log('\n' + pages.join(delimiter));
+    return this.checkStale()
+      .then(() => {
+        let endOfLine = require('os').EOL;
+        let delimiter = singleColumn ? endOfLine : ', ';
+        console.log('\n' + pages.join(delimiter));
+      });
   }
 
   printBestPage(command, options={}) {
-    const spinner = ora();
-
     // Trying to get the page from cache first
-    this.cache.getPage(command)
+    return this.cache.getPage(command)
       .then((content) => {
         // If found in first try, render it
-        if (content) {
-          this.checkStale();
-          this.renderContent(content, options);
-          return exit();
+        if (!content) {
+          // If not found, try to update
+          return spinningPromise('Page not found. Updating cache...', () => {
+            return this.cache.update();
+          })
+            .then(() => {
+              return spinningPromise('Creating index...', () => {
+                return search.createIndex();
+              });
+            })
+            .then(() => {
+              // And then, try to check in cache again
+              return this.cache.getPage(command);
+            });
```

Comment:
Let's unindent these to match the closing branch of the previous block.

Reply:
If I understood the request correctly, eslint complains about that. I deleted two leading spaces from lines 133 through 141 (inclusive), so it looks like this:

```javascript
        if (!content) {
          // If not found, try to update
          return spinningPromise('Page not found. Updating cache...', () => {
            return this.cache.update();
          })
          .then(() => {
            return spinningPromise('Creating index...', () => {
              return search.createIndex();
            });
          })
          .then(() => {
            // And then, try to check in cache again
            return this.cache.getPage(command);
          });
        }
        return content;
      })
```

This results in 8 eslint errors due to violating the "indent" rule.

Reply:
Well, that's a bit weird. But thanks for checking. :+1: